agent/xds/listeners.go

```diff
@@ -54,35 +58,225 @@ func (s *Server) listenersFromSnapshot(cInfo connectionInfo, cfgSnap *proxycfg.C
 
 // listenersFromSnapshotConnectProxy returns the "listeners" for a connect proxy service
 func (s *Server) listenersFromSnapshotConnectProxy(cInfo connectionInfo, cfgSnap *proxycfg.ConfigSnapshot) ([]proto.Message, error) {
-	// One listener for each upstream plus the public one
-	resources := make([]proto.Message, len(cfgSnap.Proxy.Upstreams)+1)
+	resources := make([]proto.Message, 1)
```

Comment:
Is there a better guesstimate that we can use here now?

Reply:
There isn't a great one. If in TProxy mode there might only be 2 listeners, since it relies on per-upstream filter chains rather than per-upstream listeners. Outside of TProxy mode it could be the size of the DiscoverChain map.

Not sure if it's worth doing having a variable starting capacity based on whether TProxy is on or not.